### ISSUE TITLE
BatchNorm Not Backward Correctly in Evaluation Mode

### ISSUE THREAD

**Issue by zhanghang1989** [apache/incubator-mxnet#9580]
## Description

I found the evaluation mode backward is already supported in the source code https://github.com/apache/incubator-mxnet/blob/master/src/operator/nn/batch_norm.cu#L368-L375, but it is not called due to the wrong context given during the execution.

## Steps to reproduce
add  ``LOG(INFO) << "SetupFlags: ctx.is_train : " << ctx.is_train;`` to SetupFlags function https://github.com/apache/incubator-mxnet/blob/master/src/operator/nn/batch_norm.cu#L578

compile MXNet with ``USE_CUDNN=0`` to make sure not calling cudnn

## Minimum reproducible example
```python
x = (mx.nd.random.uniform(0,1,shape=(B,C,H,W))).square().as_in_context(mx.gpu(0))
layer1 = nn.BatchNorm(in_channels = C)
layer1.initialize(ctx = mx.gpu(0))
with autograd.record(train_mode=False):
    y1 = layer1(x)
    loss1 = y1.sum()
loss1.backward()
```

## Output from the terminal
[22:51:47] src/operator/nn/batch_norm.cu:579: SetupFlags: ctx.is_train : 0
[22:51:47] src/operator/nn/batch_norm.cu:579: SetupFlags: ctx.is_train : 1

**Comment by zhanghang1989** [comment]
Solved. Should use ``loss1.backward(train_mode=False)``
